api/app/signals/apps/api/v1/views/signal.py

```diff
@@ -270,6 +274,88 @@ def children(self, request, pk=None):
         serializer = AbridgedChildSignalSerializer(child_qs, many=True, context=self.get_serializer_context())
         return Response(serializer.data)
 
+    @action(detail=True, url_path=r'reporter_context_brief/?$')
+    def reporter_context_brief(self, request, pk=None):
+        if not settings.API_ENABLE_SIGNAL_REPORTER_CONTEXT:
+            raise Http404  # This feature is configured off
+
+        # ADD: CHECK GLOBAL CONFIG TO SEE WHETHER THIS FEATURE IS ENABLED
+        # something with API_ENABLE_REPORTER_CONTEXT (as a feature flag) -> 404
+        # Based on a user's department a signal may not be accessible.
+        signal_exists = Signal.objects.filter(id=pk).exists()
+        signal_accessible = Signal.objects.filter_for_user(request.user).filter(id=pk).exists()
+
+        if signal_exists and not signal_accessible:
+            raise PermissionDenied()
+
+        reporter = Signal.objects.get(id=pk).reporter
+        if reporter.email is None or reporter.email == '':
+            return Response({})
+        signals = Signal.objects.filter(reporter__email__iexact=reporter.email)
+
+        signal_count = signals.count()
+
+        negative_count = signals.annotate(
+            feedback_count=Count('feedback')
+        ).filter(
+            feedback_count__gte=1
+        ).annotate(
+            feedback_max_created_at=Max('feedback__created_at'),
+            feedback_max_submitted_at=Max('feedback__submitted_at')
+        ).filter(
+            feedback__is_satisfied=False,
+            feedback__submitted_at__isnull=False,
+            feedback__created_at=F('feedback_max_created_at'),
+            feedback__submitted_at=F('feedback_max_submitted_at')
+        ).count()
+
+        open_count = signals.exclude(
+            status__state__in=[workflow.GEANNULEERD, workflow.AFGEHANDELD, workflow.GESPLITST]).count()
+
+        output = {
+            'signal_count': signal_count,
+            'negative_count': negative_count,
+            'open_count': open_count,
+        }
+
+        return Response(data=output)
+
+    @action(detail=True, url_path=r'reporter_context/?$')
+    def reporter_context(self, request, pk=None):
+        if not settings.API_ENABLE_SIGNAL_REPORTER_CONTEXT:
+            raise Http404  # This feature is configured off
+
+        # Based on a user's department a signal may not be accessible.
+        signal_exists = Signal.objects.filter(id=pk).exists()
+        signal_accessible = Signal.objects.filter_for_user(request.user).filter(id=pk).exists()
+
+        if signal_exists and not signal_accessible:
+            raise PermissionDenied()
+
+        # Only interested in Signals/complaints with the same reporter as this
+        # Signal/complaint.
+        reporter = Signal.objects.get(id=pk).reporter
+        if reporter.email is None or reporter.email == '':
+            return Response({})
+
+        feedback_qs = Feedback.objects.order_by('-created_at')
+        signal_qs = (
```

Comment:
You are using a query for the prefetch related? I think this is not needed because we already have a feedback connection through the foreingkey on the Feedback model that is accessible by using the `feedback` related_name defined on the Feedback model. Therefore a `.prefetch_related('feedback')` should do the trick.